src/scenic/core/sample_checking.py

```diff
@@ -0,0 +1,108 @@
+"""The SampleChecker class and it's implementations."""
+
+from abc import ABC, abstractmethod
+import time
+
+from scenic.core.requirements import IntersectionRequirement
+
+class SampleChecker(ABC):
+    def __init__(self):
+        self.requirements = None
+
+    def addRequirements(self, requirements):
```

Comment:
Let's call this `setRequirements` since the current name suggests that you could call it multiple times and accumulate requirements.